src/compile/scale/assemble.ts

```diff
@@ -5,17 +5,21 @@ import {stringValue, vals} from '../../util';
 import {isDataRefDomain, isDataRefUnionedDomain, isFieldRefUnionDomain, isSignalRefDomain, VgDataRef, VgScale} from '../../vega.schema';
 import {Model} from '../model';
 import {isRawSelectionDomain, selectionScaleDomain} from '../selection/selection';
+import {mergeDomains} from './domain';
 
 
 export function assembleScale(model: Model): VgScale[] {
-    return vals(model.component.scales).reduce((scales: VgScale[], scaleComponent) => {
+    return vals(model.component.scales).reduce((scales, scaleComponent) => {
       if (scaleComponent.merged) {
         // Skipped merged scales
         return scales;
       }
 
       // We need to cast here as combine returns Partial<VgScale> by default.
-      const scale = scaleComponent.combine(['name', 'type', 'domain', 'domainRaw', 'range']) as VgScale;
+      const scaleComProps = scaleComponent.combine(['name', 'type', 'domainRaw', 'range']);
```

Comment:
Instead of passing `['name', 'type', 'domainRaw', 'range']` to combine.  

Maybe do this below to preserve the nice scale order:
```
const {name, type, domainRaw, range, ...otherScaleProps} = scale;
scales.push({
  name,
  type,
  ...(domain ? {domain} : {}),
  ...(domainRaw ? {domainRaw} : {}),
  ...(range ? {range} : {}),
})
```

I think we can also remove the optional parameter from the `combine` method as I believe it is only used here.